ecdh_test.go

```diff
@@ -0,0 +1,58 @@
+package ecdh_test
+
+import (
+	"crypto/aes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/watnet/ecdh"
+)
+
+func TestEncrypt(t *testing.T) {
+	tests := []struct {
+		name string
+		data []byte
+	}{
+		{
+			name: "NoPadding",
+			data: []byte("testbutnopadding"),
+		},
+		{
+			name: "Padding",
+			data: []byte("testwithpadding"),
+		},
+		{
+			name: "LongWithPadding",
+			data: []byte("this is some text that is longer than a single block and has padding"),
+		},
+	}
+
+	for _, test := range tests {
+		test := test
```

Comment:
What's this line for? Isn't it redundant?

Reply:
No. The `test` declared in the `for` itself using `range` is in a scope between the outside of the loop and the inside. It's the same variable every time. Because the tests run in parallel, it causes a data race. The line in question creates a new variable _inside_ of the loop that shadows the other one.